sklearn/tree/tests/test_tree.py

```diff
@@ -2551,22 +2601,24 @@ def test_missing_values_is_resilience(
     else:
         sample_weight = None
 
-    native_tree = Tree(max_depth=10, random_state=global_random_seed)
```

Comment:
Is there an issue with max_depth=10 or is max_depth=None better here?

Reply:
No there is no issue. I think I removed it awhile ago, but I can add it back in as I guess it would make the trees fit faster, but in the unit-test, imo it doesn't make sense.

I.e. we don't provide motivation for why the max_depth is limited. 

I added an in-line comment specifying this is purely for making the test go a bit faster.

Reply:
See: [0753aa2](https://github.com/scikit-learn/scikit-learn/pull/27966/commits/0753aa25b81feaeed1b9f94b3aeb54863a4a567e)